mixer/test/listbackend/list.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// nolint:lll
+//go:generate go run $GOPATH/src/istio.io/istio/mixer/tools/mixgen/main.go adapter -n listbackend-nosession -c $GOPATH/src/istio.io/istio/mixer/adapter/list/config/config.proto_descriptor -s=false -t listentry -o nosession.yaml
+
+package listbackend
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net"
+	"sync"
+
+	"google.golang.org/grpc"
+
+	adptModel "istio.io/api/mixer/adapter/model/v1beta1"
+	"istio.io/istio/mixer/adapter/list"
+	"istio.io/istio/mixer/adapter/list/config"
+	"istio.io/istio/mixer/pkg/adapter"
+	"istio.io/istio/mixer/pkg/pool"
+	"istio.io/istio/mixer/pkg/runtime/handler"
+	"istio.io/istio/mixer/template/listentry"
+)
+
+type (
+	// Server is basic server interface
+	Server interface {
+		Addr() net.Addr
+		Close() error
+		Run()
+		Wait() error
+	}
+
+	// nosessionServer models no session adapter backend.
+	nosessionServer struct {
+		listener net.Listener
+		shutdown chan error
+		server   *grpc.Server
+
+		rawcfg      []byte
+		builder     adapter.HandlerBuilder
+		env         adapter.Env
+		builderLock sync.RWMutex
```

Comment:
It'd be more helpful is this were called handlerLock, since that's what is being protected.

Reply:
D